extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -548,6 +555,22 @@ export class SafeframeHostApi {
     }
   }
 
+  /**
+   * @param {!JsonObject} payload
+   * @private
+   */
+  handleShrinkRequest_(payload) {
+    if (!this.isRegistered_) {
+      return;
+    }
+    const shrinkHeight = Number(this.iframe_.height) -
+          payload['shrink_b'] + payload['shrink_t'];
+    const shrinkWidth = Number(this.iframe_.width) -
+          payload['shrink_r'] + payload['shrink_l'];
+    this.resizeAmpAdAndSafeframe(shrinkHeight, shrinkWidth,
```

Comment:
What happens if shrink payload values are not present or numbers?  What about passing a size that's actually larger than the creative or is negative?

Reply:
Done